packages/nimble-components/src/wafer-map/modules/rendering.ts

```diff
@@ -1,60 +1,60 @@
 import type { WaferMap } from '..';
-import { DieRenderInfo, Dimensions, HoverDieOpacity } from '../types';
+import { DieRenderInfo, HoverDieOpacity } from '../types';
 
 /**
  * Responsible for drawing the dies inside the wafer map, adding dieText and scaling the canvas
  */
 export class RenderingModule {
-    private readonly dies: DieRenderInfo[];
-    private readonly dimensions: Dimensions;
-    private readonly labelFontSize: number;
+    private dies!: DieRenderInfo[];
+    private readonly minDieDim = 50;
 
-    public constructor(private readonly wafermap: WaferMap) {
-        this.dies = wafermap.dataManager!.diesRenderInfo;
-        this.dimensions = wafermap.dataManager!.dieDimensions;
-        this.labelFontSize = wafermap.dataManager!.labelsFontSize;
+    public constructor(wafermap: Readonly<WaferMap>) {

```

Comment:
Why are so many of these changed to pass the wafer map to every function call instead of just saving the reference like it was before?

You wouldn't do the following right?:
```
const map1 = new WaferMap()
const map2 = new WaferMap();
const renderingModule = new RenderingModule(map1);
renderingModule.drawWafer(map2);
```

So why is this change allowing that to happen?

Reply:
Before, each time the wafer was updated the event handlers, the data manager and  the rendering module was reinstated, so I was moving towards having them instantiated once in the connected callback and whenever called from update it will just update the needed attributes of the modules or the wafermap . If that was the case I thought that having the local reference of the wafermap attributes will keep them out of sync with the component, so passing the reference of the main component to the function and using it to update the model or the component attributes directly will be best . If that is not the case, are there any best practices I should use?

Reply:
> If that was the case I thought that having the local reference of the wafermap attributes will keep them out of sync with the component,

I don't know if I understand the concern. I created the following examples to try and understand. What was previously there was the pattern in class A. Some attributes of the wafermap were saved off (some weren't, it was inconsistent). You want to switch to B where no attributes are saved off but you need to pass the wafermap to every method call. I'm suggesting doing C where no attributes are saved off, you just hold a reference to the wafermap. Is there something I'm missing in C that causes attributes to get out of sync?

```
class A {
   constructor (wafermap) {
       this.val = wafermap.val;
   }
   method () {
       this.val
    }
```

```
class B {
   constructor (wafermap) {
       wafermap.val;
   }
   method (wafermap) {
       wafermap.val
    }
```

```
class C {
   constructor (wafermap) {
       this.wafermap = wafermap
       this.wafermap.val;
   }
   method () {
       this.wafermap.val
    }
```

Reply:
I think in Case C they should all work, i will try it out and test if there are no problems